chromium_src/components/autofill/core/browser/autofill_browser_util.cc

```diff
@@ -0,0 +1,33 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_CHROMIUM_SRC_COMPONENTS_AUTOFILL_CORE_BROWSER_AUTOFILL_BROWSER_UTIL_CC_
+#define BRAVE_CHROMIUM_SRC_COMPONENTS_AUTOFILL_CORE_BROWSER_AUTOFILL_BROWSER_UTIL_CC_
+
+#define IsFormMixedContent IsFormMixedContent_ChromiumImpl
+
+#include "src/components/autofill/core/browser/autofill_browser_util.cc"
+#undef IsFormMixedContent
+
+namespace {
+constexpr char kOnionDomain[] = "onion";
+}  // namespace
+
+namespace autofill {
+
+bool IsFormMixedContent(const AutofillClient& client, const FormData& form) {
+  if (IsFormMixedContent_ChromiumImpl(client, form)) {
+    return true;
+  }
+
+  return client.GetLastCommittedPrimaryMainFrameOrigin().DomainIs(
```

Comment:
i would have assumed that an HTTP form is mixed content if any of its frame ancestors are secure origins, not just the top level origin. however it looks like the chromium implementation also only checks the top level? https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/ui/autofill/chrome_autofill_client.cc;l=903;drc=4c6d5b3c3ed16fa75258c6b4763188f840cb433d;bpv=1;bpt=1

it may make more sense to override https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/ui/autofill/chrome_autofill_client.cc;l=903;drc=4c6d5b3c3ed16fa75258c6b4763188f840cb433d;bpv=1;bpt=1 if we think this could affect other places in autofill.

Reply:
Maybe it's because an [HTTP iframe on an HTTPS page](https://fmarier.org/unprotected/insecure-iframe.html) would get blocked as mixed content anyways?

Reply:
> Maybe it's because an [HTTP iframe on an HTTPS page](https://fmarier.org/unprotected/insecure-iframe.html) would get blocked as mixed content anyways?

is that the case for http frames on onion pages as well?

Reply:
Yes: http://ixrdj3iwwhkuau5tby5jh3a536a2rdhpbdbu6ldhng43r47kim7a3lid.onion/unprotected/insecure-iframe.html